test/common/http/utility_test.cc

```diff
@@ -574,6 +574,31 @@ TEST(HttpUtility, TestParseCookieWithQuotes) {
   EXPECT_EQ(Utility::parseCookieValue(headers, "leadingdquote"), "\"foobar");
 }
 
+TEST(HttpUtility, TestParseCookieMultipleValues) {
+  TestRequestHeaderMapImpl headers{{"cookie", "somekey=foo"},
```

Comment:
I would also add test with cookies that contain different keys and also negative test where none of the keys match.

Reply:
Added tests with https://github.com/envoyproxy/envoy/pull/16344/commits/5406686119624499e52d0d007e351ec06348200f . Did you have other things in mind?